app/helpers/form_helper.py

```diff
@@ -94,13 +96,22 @@ def get_mapped_answers(schema, answer_store, location):
     Maps the answers in an answer store to a dictionary of key, value answers.
     """
 
-    block_id = location.block_id
-
-    answer_ids = schema.get_answer_ids_for_block(block_id)
-
-    answers = answer_store.get_answers_by_answer_id(
-        answer_ids=answer_ids, list_item_id=location.list_item_id
-    )
-    result = {answer.answer_id: answer.value for answer in answers if answer}
+    answer_ids = schema.get_answer_ids_for_block(location.block_id)
+    result = {}
+
+    if isinstance(location, RelationshipLocation):
+        answer = answer_store.get_answer(answer_ids[0])
```

Comment:
As discussed IRL, the only operation we do on the result is to iterate over it. So this should be converted to a list.

Reply:
I hadn't quite realised the significance of the set being used as question variants will have duplicate answer ids. However, as the answer ids must be the same across all question variants, this seems unnecessary. I'll look into removing the set and getting the answer ids from the first variant.

Reply:
I've added a `get_relationship_answer_id_for_block` method to `QuestionnaireSchema` so that we don't need access the list in this way here. I've also removed the use of of sets as they weren't necessary.